Assets/LeapMotionModules/UI/Scripts/Constraints/HingeConstraint.cs

```diff
@@ -0,0 +1,196 @@
+﻿using Leap.UI.Constraints;
+using Leap.Unity.Attributes;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.UI.Constraints {
+
+  public class HingeConstraint : AngularRotationConstraint {
+
+    [Header("Lever Configuration")]
+    [Tooltip("Whether the lever's angular range of motion should extend from its beginning, center, or end.")]
+    public Anchor leverAnchor = Anchor.Center;
+    [Tooltip("The local-space axis around which the lever allows rotation.")]
+    public Axis leverAxis = Axis.X;
+    [Tooltip("The total angular width the lever is allowed to rotate.")]
+    [MinValue(0F)]
+    [MaxValue(360F)]
+    public float angularWidth = 60F;
+
+    public override Quaternion EvaluateRotationAlongConstraint(float angleAlongConstraint) {
+      float axisSign = leverAnchor == Anchor.End ? -1 : 1;
+      return Quaternion.AngleAxis(angleAlongConstraint, axisSign * (leverAxis.ToUnitVector3()));
+    }
+
+    public override float GetConstraintProjectionAngle() {
+      // Normalize the local rotation.
+      transform.localRotation = Quaternion.Lerp(transform.localRotation, this.transform.localRotation, 1F);
+
+      // Find angle by projection and trig.
+      Vector3 localLeverAxis = constraintLocalRotation * leverAxis.ToUnitVector3();
+      Vector3 localProjection = Vector3.ProjectOnPlane(transform.localRotation * ProjectionDirection, localLeverAxis).normalized;
+      if (localProjection.magnitude < 0.0001F) {
+        return 0F;
+      }
+      else {
+        Vector3 zeroAngleProjection = constraintLocalRotation * ProjectionDirection;
+        float angle = Vector3.Angle(localProjection, zeroAngleProjection);
+        if (Vector3.Dot(localLeverAxis, Vector3.Cross(zeroAngleProjection, localProjection)) > 0F) {
+          // positive angle
+        }
+        else {
+          // negative angle
+          angle = -angle;
+        }
+        if (leverAnchor == Anchor.Beginning) {
+          float cutoffAngle = (MinAngle + (MaxAngle - 360F)) / 2F;
+          if (angle < cutoffAngle) angle += 360F;
+          return Mathf.Clamp(angle, MinAngle, MaxAngle);
+        }
+        else if (leverAnchor == Anchor.End) {
+          angle = -angle;
+          if (angle < 0F) {
+            angle += 360F;
+          }
+          float cutoffAngle = 180F + (MaxAngle + (MinAngle)) / 2F;
+          if (angle > cutoffAngle) {
+            angle -= 360F;
+          }
+          return Mathf.Clamp(angle, MaxAngle, MinAngle);
+        }
+        else {
+          return Mathf.Clamp(angle, MinAngle, MaxAngle);
+        }
+      }
+    }
+
+    private float ComponentOf(Vector3 vector, Axis axis) {
+      switch (axis) {
+        case Axis.X:
+          return (vector.y == 180F && vector.z == 180F ? 180F : 0F) - vector.x * (vector.y == 180F && vector.z == 180F ? 1 : -1F);
```

Comment:
Because euler angles sometimes return the angle you're looking for but other times it returns the opposite of the angle you're looking for with -180 as the Y and Z components. However, I've removed this getter because it was for an older, less-reliable constraint calculation method, as well as other properties I wrote but that were never used.